src/FindMostExpensiveSet.java

```diff
@@ -0,0 +1,56 @@
+import java.util.List;
+
+public class FindMostExpensiveSet {
+    private int backpackVolume = 15;
+    private List<Item> resultSet;
+    private int resultWeight = 0;
+
+    public void findSet(List<Item> entrySet) {
+        if (entrySet.isEmpty()) {
+            return;
+        }
+
+        isMostExpensive(entrySet);
+
+        for (int i = 0; i <entrySet.size(); i++) {
+            List<Item> temp = entrySet;
```

Comment:
следует создавать копию списка, а не модифицировать текущий.
```suggestion
            List<Item> temp = new ArrayList<>(entrySet);
```